frontend/src/metabase/entities/actions/actions.ts

```diff
@@ -201,6 +206,50 @@ const Actions = createEntity({
   actions: {
     enableImplicitActionsForModel,
   },
+  objectActions: {
+    createPublicLink: createAction(
+      CREATE_PUBLIC_LINK,
+      ({ id }: { id: number }) => {
+        return ActionsApi.createPublicLink({ id }).then(
+          ({ uuid }: { uuid: string }) => {
+            return {
+              id,
+              uuid,
+            };
+          },
+        );
+      },
+    ),
+    deletePublicLink: createAction(
+      DELETE_PUBLIC_LINK,
+      ({ id }: { id: number }) => {
+        return ActionsApi.deletePublicLink({ id }).then(() => {
+          return {
+            id,
+          };
+        });
+      },
+    ),
+  },
```

Comment:
Should we use `async/await` here?

Reply:
TBH, I don't see much improvement from converting that to async/await here. In certain places, when dealing with lots of nested promise hell that could be a good reason to do async/await. But they're almost equal in my opinion here, I like using promise by default because when dealing with `catch` block it's easier and less verbose to write `.catch` than to write `try/catch` imo. What do you think?

ps. one interesting note is that VS code always prompt me to convert this to async/await, maybe it favor one over another 😃